linux/shim/init.go

```diff
@@ -60,15 +62,27 @@ func newInitProcess(context context.Context, path, namespace string, r *shimapi.
 		}
 		options = *v.(*runcopts.CreateOptions)
 	}
+
+	rootfs := filepath.Join(path, "rootfs")
+	// count the number of successful mounts so we can undo
+	// what was actually done rather than what should have been
+	// done.
+	nrRootMounts := 0
```

Comment:
`EINVAL` covers a multitude of sins according to [umount(2)](http://man7.org/linux/man-pages/man2/umount.2.html), including "bad flags" in addition to "not a mount point" (I'm not too worried about "called with MNT_EXPIRE and either MNT_DETACH or MNT_FORCE").

I think it's still the right way to go though.